modules/node-shared/src/main/scala/org/tessellation/node/shared/infrastructure/security/DoubleSignDetectProgram.scala

```diff
@@ -0,0 +1,34 @@
+package org.tessellation.node.shared.infrastructure.security
+
+import cats.effect.Async
+import cats.effect.std.Supervisor
+import cats.syntax.all._
+
+import scala.concurrent.duration.FiniteDuration
+
+import org.tessellation.node.shared.domain.Daemon
+import org.tessellation.node.shared.domain.fork.ForkInfoStorage
+import org.tessellation.node.shared.domain.security.{DoubleSignDetect, DoubleSignDetectRunner}
+
+import org.typelevel.log4cats.slf4j.Slf4jLogger
+
+object DoubleSignDetectProgram {
+
+  def make[F[_]: Async](storage: ForkInfoStorage[F], doubleSignValidator: DoubleSignDetect[F]): DoubleSignDetectRunner[F] =
+    new DoubleSignDetectRunner[F] {
+
+      val logger = Slf4jLogger.getLogger
+
+      def run: F[Unit] = storage.getForkInfo.flatMap {
```

Comment:
Does "getForkInfo" ensure ordering? I'd use SortedSet to make sure

Reply:
`getForkInfo` returns a `F[ForkInfoMap]`. `ForkInfoMap` is `Map[PeerId, ForkInfoEntries]`. `ForkInfoEntries` is a case class around `PosInt` and `Queue[ForkInfo]`. So, yes, ordering is ensured.